compiler/src/dmd/expressionsem.d

```diff
@@ -10011,17 +10011,22 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             (ae.e2.type.ty == Tsarray || ae.e2.type.ty == Tarray) &&
             (ae.e1.type.nextOf && ae.e2.type.nextOf && ae.e1.type.nextOf.mutableOf.equals(ae.e2.type.nextOf.mutableOf));
 
-        if (!isArrayAssign)
+        const isArraySetAssign =
+            (ae.e1.isSliceExp || ae.e1.type.ty == Tsarray) &&
```

Comment:
Please use the call operator `()` for function calls so that it is easier to differentiate between fields and function calls.